### DISCUSSION TITLE
$watch method run twice in inputs field

### DISCUSSION THREAD

**Discussion by ousid** [alpinejs/alpine#3238] [7. Bugs (archived)]
AlpineJS V: `3.10.5`

## Steps to produce

HTML:

```html
<input x-ref="input" x-model="input" x-ref="input" type="text"/>
```

JS file:

```js
...
init() {
   this.$watch('input', query => {
    console.log(query);
  });
}
...
```

After running this code, you'll notice that the input value `query` will show twice in a row, and the issue I'm facing, that I'm sending the query to the server side, and it causes 2 queries, that it's not good for the performance.

Have you faced this issue before, or know how to solve it?

Thanks in advance.

**Comment by ekwoka** [comment]
You can debounce it

**Reply by ousid** [reply]
If you are referring to the `x-model` directive, I have done it already, and the same results.

**Reply by SimoTod** [reply]
3.4.2 is old. The latest version is 3.10.5. Have you checked that version? If there's really a bug, it will be fixed on the latest version anyway.

Also, double check you are not including / starting alpine twice (this usually has side effects such as duplicate event listeners, for loops and, I believe, $watch callbacks)

**Reply by ousid** [reply]
Thanks, @SimoTod 

I have checked all of that after I created this discussion. And yeah, I'm running the latest version of `alpine`, besides I searched for a duplicate, but I don't think I have one.

I'll test this logic outside my current project and see.

Thanks again.
